src/Authentication/LoginHandler.php

```diff
@@ -0,0 +1,138 @@
+<?php
+
+namespace Bugcache\Authentication;
+
+use Aerys\Bootable;
+use Aerys\Logger;
+use Aerys\ParsedBody;
+use Aerys\Request;
+use Aerys\Response;
+use Aerys\Server;
+use Aerys\ServerObserver;
+use Aerys\Session;
+use Amp\Promise;
+use Amp\Success;
+use Bugcache\Mustache;
+use Bugcache\SessionKeys;
+use Bugcache\Storage\AuthenticationRepository;
+use Bugcache\Storage\UserRepository;
+use function Aerys\parseBody;
+use function Amp\resolve;
+
```

Comment:
I'm using 2016.1, the only main red errors left are when doing `($this->handler)($req)` (fcalls with function in parens).

Often enough to check the origin of a class name